src/__experimental__/components/form/form.component.js

```diff
@@ -128,16 +134,28 @@ class FormWithoutValidations extends React.Component {
   }
 
   addUnsavedWarningListener = () => {
+    this.unsavedListener = true;
     this._window.addEventListener('beforeunload', this.checkIsFormDirty);
   }
 
   removeUnsavedWarningListener = () => {
+    this.unsavedListener = false;
     this._window.removeEventListener('beforeunload', this.checkIsFormDirty);
   }
 
+  /** Returns true if any input has value */
+  checkFormDataExists() {
+    const { formInputs } = this.state;
+    return Object.keys(formInputs).map(id => (!!formInputs[id].length)).includes(true);
```

Comment:
Just a minor comment: I think this can be simplified using [`Array.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). Also, ESLint's [`no-implicit-coercion`](https://eslint.org/docs/rules/no-implicit-coercion) rule recommends using `Boolean(foo)` instead of `!!foo` (I think it'd be good to enable that rule).
```js
return Object.keys(formInputs).some(id => Boolean(formInputs[id].length));
```

Reply:
No problem I will make the change and I'll add the rule update to the es-lint ticket raised the other day

Reply:
Many thanks for refactoring this and updating FE-1870 :white_check_mark: